src/tasks/AotCompilerTask/MonoAOTCompiler.cs

```diff
@@ -271,6 +281,44 @@ private bool ProcessAndValidateArguments()
         if (!Directory.Exists(IntermediateOutputPath))
             Directory.CreateDirectory(IntermediateOutputPath);
 
+        if (!string.IsNullOrEmpty(NetTracePath))
+        {
+            if (!File.Exists(NetTracePath))
+            {
+                Log.LogError($"NetTracePath={NetTracePath} doesn't exist");
+                return false;
+            }
+            if (!File.Exists(PgoBinaryPath))
+            {
+                Log.LogError($"NetTracePath was provided, but {nameof(PgoBinaryPath)}='{PgoBinaryPath}' doesn't exist");
+                return false;
+            }
+
+            var OutputMibcPath = Path.Combine(OutputDir, Path.ChangeExtension(Path.GetFileName($"{NetTracePath}"), ".mibc"));
+            (int exitCode, string output) = Utils.TryRunProcess(Log,
+                                                                PgoBinaryPath,
+                                                                $"create-mibc --trace {NetTracePath} --output {OutputMibcPath}",
```

Comment:
Emit the output a temporary file, then use `if (Utils.CopyIfDifferent(tmpPath, finalPath, useHash: true))` to copy to the actual output path. This is needed for incremental builds.

Reply:
For incremental builds, wouldn't the point be to avoid overwriting files if it has the same content as before?
I'm guessing then we would want to avoid overwriting the `OutputMibcPath`? Or do we actually want to track any changes to what this process emits into the console?

If the former, the dotnet-gpo tool generates the .mibc file, so if we were to generate the .mibc in a temporary path and compare contents with the final `.mibc` file path, we would want to delete the file afterwards right? 

If the latter, it wouldn't really affect `OutputMibcPath` itself right?

I'm not quite sure how this helps, as the tool will be ran anyways?

Reply:
> For incremental builds, wouldn't the point be to avoid overwriting files if it has the same content as before? I'm guessing then we would want to avoid overwriting the `OutputMibcPath`? Or do we actually want to track any changes to what this process emits into the console?

Yep, and `if (Utils.CopyIfDifferent(tmpPath, OutputMibcPath, useHash: true))` - will copy to `OutputMibcPath` only if the files are different.

See https://github.com/dotnet/runtime/blob/main/src/tasks/AotCompilerTask/MonoAOTCompiler.cs#L1001-L1005

And you can delete the temp file 👍

Reply:
I've ran the sample a few times, and each time it seems like `Utils.CopyIfDifferent(tmpMibcPath, outputMibcPath, useHash: false)` evaluates to true. not sure what inherently is different two mibc files created from the same .nettrace is.

Reply:
useHash:true

Reply:
Looks like the .mibc file generated will include the filename itself. So in this case it will always have different contents unfortunately.

Reply:
![Screen Shot 2022-06-17 at 12 27 55 PM](https://user-images.githubusercontent.com/16830051/174338869-773c4e50-e191-4ffb-b28c-d4643526749f.png)

Reply:
Resolved offline, opting for a cache option to check whether or not the .nettrace file being used to generate the .mibc is the same (i.e. if so, then avoid regenerating the .mibc).